pkg/ruler/remotequerier.go

```diff
@@ -297,6 +298,9 @@ func (q *RemoteQuerier) sendRequest(ctx context.Context, req *httpgrpc.HTTPReque
 		if err == nil {
 			return resp, nil
 		}
+		if code := grpcutil.ErrorToStatusCode(err); code/100 == 4 {
```

Comment:
[nit] Please add a short comment to explain *why* we do it. Something like this:

```
4xx errors shouldn't be retried because we expect to hit the same exact error again if we'll run the same query again.
```